crates/ruff/src/message/text.rs

```diff
@@ -180,23 +188,53 @@ impl Display for MessageCodeFrame<'_> {
         let source_code = file.to_source_code();
 
         let content_start_index = source_code.line_index(range.start());
+        let content_start_cell = self
+            .jupyter_index
+            .and_then(|jupyter_index| jupyter_index.cell(content_start_index.get()));
         let mut start_index = content_start_index.saturating_sub(2);
 
-        // Trim leading empty lines.
+        // Trim leading empty lines or skip lines outside of the content cell
+        // for jupyter notebooks.
         while start_index < content_start_index {
+            if let Some(jupyter_index) = self.jupyter_index {
```

Comment:
Should we put this outside of the `while`, and have separate `while` loops in the `if` and `else` branches? The control flow here is a little hard to follow.

Reply:
Make sense. I've separated them out and also used `unwrap_or_default` for `index.cell` and `index.cell_row`. The unwrap failure is unlikely but if it happens then the rendering will be incorrect and would most likely be reported by the user.